### PR TITLE
KOGITO-3073 Create protobuf common module to allow reusing marshaller…

### PR DESCRIPTION
**Pull request by cristianonicolai**
…s outside Infinispan addon

Moving shared code to deal with protobuf files into a common module. This will allows new persistence types to reuse it if needed.

### CODE REVIEW

**Comment by cristianonicolai** [comment]
jenkins please retest this

File: addons/persistence/infinispan-persistence-addon/src/main/java/org/kie/kogito/infinispan/CacheProcessInstances.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import org.infinispan.client.hotrod.RemoteCache;
 import org.infinispan.client.hotrod.RemoteCacheManager;
 import org.infinispan.protostream.BaseMarshaller;
+import org.kie.kogito.persistence.protobuf.ProtoStreamObjectMarshallingStrategy;
```
Comment (tiagodolphine) [inline addons/persistence/infinispan-persistence-addon/src/main/java/org/kie/kogito/infinispan/CacheProcessInstances.java:25]: is this import needed?

**Review by tiagodolphine — APPROVED** [review]
LGTM just one minor comment.
Thanks.

File: addons/persistence/infinispan-persistence-addon/src/main/java/org/kie/kogito/infinispan/CacheProcessInstances.java (hunk 1 above)
Comment (cristianonicolai) [reply addons/persistence/infinispan-persistence-addon/src/main/java/org/kie/kogito/infinispan/CacheProcessInstances.java:25]: yep, moved the class to a different package in the common module.

**Review by MarianMacik — APPROVED** [review]

**Pull request merged**